autofit/graphical/expectation_propagation/factor_optimiser.py

```diff
@@ -31,7 +31,10 @@ def exact_fit(
         self, factor_approx: FactorApproximation, status: Status = Status()
     ) -> Tuple[MeanField, Status]:
         factor = factor_approx.factor
-        cavity_dist = factor_approx.cavity_dist
+        cavity_dist = factor_approx.cavity_dist.copy()
+        for v in factor_approx.mean_field.keys() - cavity_dist:
+            cavity_dist[v] = factor_approx.mean_field[v].zeros_like()
```

Comment:
@rhayes777 

For `exact_fit`s of `FactorApproximation`s we need the full cavity_dist, so in this case we need to create `zeros_like` versions of the missing `cavity_dist`s - e.g. for the Normal Distribution the zeros_like dist has infinite variance, note that its logpdf will be `- inf` so we don't want to evaluate it in general.

Reply:
Interesting. Could you write something along those lines in the code docs?